java/jdbc/src/main/java/io/vitess/jdbc/VitessVTGateManager.java

```diff
@@ -174,45 +178,56 @@ private static void actuallyCloseRefreshedConnection(final VTGateConnection old)
     }
   }
 
+
+  private static String nullIf(String ifNull, String returnThis) {
+    if (ifNull == null) {
+      return returnThis;
+    } else {
+      return ifNull;
+    }
+  }
+
   /**
    * Create vtGateConn object with given identifier.
    */
   private static VTGateConnection getVtGateConn(VitessJDBCUrl.HostInfo hostInfo,
-      VitessConnection connection) {
+                                                VitessConnection connection) {
     final Context context = connection.createContext(connection.getTimeout());
     RetryingInterceptorConfig retryingConfig = getRetryingInterceptorConfig(connection);
     if (connection.getUseSSL()) {
-      final String keyStorePath = connection.getKeyStore() != null ? connection.getKeyStore()
-          : System.getProperty(Constants.Property.KEYSTORE_FULL);
-      final String keyStorePassword =
-          connection.getKeyStorePassword() != null ? connection.getKeyStorePassword()
-              : System.getProperty(Constants.Property.KEYSTORE_PASSWORD_FULL);
-      final String keyAlias = connection.getKeyAlias() != null ? connection.getKeyAlias()
-          : System.getProperty(Constants.Property.KEY_ALIAS_FULL);
-      final String keyPassword = connection.getKeyPassword() != null ? connection.getKeyPassword()
-          : System.getProperty(Constants.Property.KEY_PASSWORD_FULL);
-      final String trustStorePath = connection.getTrustStore() != null ? connection.getTrustStore()
-          : System.getProperty(Constants.Property.TRUSTSTORE_FULL);
-      final String trustStorePassword =
-          connection.getTrustStorePassword() != null ? connection.getTrustStorePassword()
-              : System.getProperty(Constants.Property.TRUSTSTORE_PASSWORD_FULL);
-      final String trustAlias = connection.getTrustAlias() != null ? connection.getTrustAlias()
-          : System.getProperty(Constants.Property.TRUST_ALIAS_FULL);
-
-      final TlsOptions tlsOptions = new TlsOptions().keyStorePath(keyStorePath)
-          .keyStorePassword(keyStorePassword).keyAlias(keyAlias).keyPassword(keyPassword)
-          .trustStorePath(trustStorePath).trustStorePassword(trustStorePassword)
-          .trustAlias(trustAlias);
-
-      return new RefreshableVTGateConnection(
-          new GrpcClientFactory(retryingConfig).createTls(context, hostInfo.toString(), tlsOptions),
-          keyStorePath, trustStorePath);
+      TlsOptions tlsOptions = getTlsOptions(connection);
+      RpcClient rpcClient = new GrpcClientFactory(retryingConfig)
+          .createTls(context, hostInfo.toString(), tlsOptions);
+      String path = tlsOptions.getTrustStore().getPath();
```

Comment:
The path variable is not needed. Just create this in the construction invocation like the other path `tlsOptions.getKeyStore().getPath()`